fbpic/lpa_utils/laser/longitudinal_laser_profiles.py

```diff
@@ -183,3 +184,127 @@ def squared_profile_integral(self):
         See the docstring of LaserLongitudinalProfile.squared_profile_integral
         """
         return (0.5 * np.pi * 1./self.inv_ctau2)**.5
+
+
+class CustomSpectrumLongitudinalProfile(LaserLongitudinalProfile):
+    """Class that calculates a longitudinal profile with a user defined
+    spectral amplitude and phase."""
+
+    def __init__(self, z0, spectrum_file, propagation_direction=1):
+        """
+        Define the complex longitudinal profile of the laser pulse,
+        from the spectrum provided in `spectrum_file`.
+
+        More specifically, the temporal characteristics of the pulse are
+        calculated numerically via the spectral intensity and phase which
+        are provided to the class as a path to a csv file containing the data.
+
+        More specifically, the electric field computed according to:
+        .. math::
+
+            E(z,t) \propto \int_{-\infty}^{\infty}\!\! dk \;k
+                \sqrt{\\frac{dI}{d\lambda}(2\pi/k)}
+                \,e^{i\phi(2\pi/k)}\,e^{i k (z-z0 -ct)}
+
+        where :math:`\\frac{dI}{d\lambda}` and :math:`\phi` are the
+        spectral intensity provided in the csv file (as a function of the
+        wavelength :math:`\lambda=2\pi/k`). (The fact that the integrand
+        is multiplied by :math:`k` in the above formula is because
+        the csv file provides :math:`\\frac{dI}{d\lambda}`, not
+        :math:`\\frac{dI}{dk}`.)
```

Comment:
```suggestion
        the csv file provides the intensity distribution over wavelengths 
        :math:`\lambda`, instead of over wavevectors :math:`k`.)
```